ahdilmore/metagenomics_pooling_notebook
# 18S and ITS updates (#38)

* 18S and ITS amplicon updates

modified amplipool.py and amplicon notebook; Run all cells works

* Update amplipool.py

checked with flake8 to make sure appropriate spaces

* Update amplicon-pooling.ipynb

set seqtype to 16S

* Update metapool/amplipool.py

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>

* update for test_amplipool.py

* Update test_amplipool.py

* remove xlsx files

* Update test_amplipool.py

I think I maybe understand what needed to be changed for test_amplipool.py to run appropriately. But, maybe not.

* Create template.xlsx

* add new function generate_qiita_prep_file

hides all of the other code

* Update prep.py

string lengths under control

* update testing

let's see if this works

* fixed other problems with testing

maybe finished

* more test fixes

maybe this time?

* update test_amplipool and test_prep

I think the last batch of errors should be fixed, but doubtful there is more

* more updates to test_amplipool and test_prep

Trying to fix things...

* back at it again, updates to test_amplipool and test_prep

sigh, maybe this time?

* Update test_amplipool.py

finally?!?

* Update test_amplipool.py

maybe?

* Update test_amplipool.py

PCR segments should be the same now

* Update primers for amplipool.py

I don't know anymore

* More adjustments to primers in test_amplipool

hopefully this time

* Try again with test_amplipool

copy and pasted primer sequences so they _should_ be right

* Update so hopefully everything matches now in test_amplipool

Meh

* copy and pasted better in test_amplipool

sigh

* Ordered correctly on test_amplipool

I need to go to bed and stop making dumb errors, but I feel like I'm so close to that green check mark!

* last one for now, update test_amplipool

* update test_amplipool for 18S

I need to stop

* modified test_amplipool

for each amplicon type, better

* Modified golay barcodes for test_amplipool

* update primers for data2 in test_amplipool

Okay, not sure why the primers are not the same, but whatever

* further updates to test_amplipool

numbering system not like I thought, suspect the renaming of things in the upper portion is the culprit.

* another fix for test_amplipool

fixed golay barcodes that were not in the expected order for data3. I think we're almost done.

* more primer updates to test_amplipool

update primers for data3. PCR primers are all different? I guess because they include the barcodes?

* Update prep.py

* rm DS_store

* Delete .DS_Store

* Update metapool/tests/test_amplipool.py

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>

* Update metapool/tests/test_amplipool.py

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>

* Update amplicon-pooling.ipynb

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>

* Update metapool/tests/test_prep.py

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>

* Update metapool/tests/test_amplipool.py

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>

Co-authored-by: Yoshiki Vázquez Baeza <yoshiki@ucsd.edu>
